extensions/amp-social-share/0.1/amp-social-share.js

```diff
@@ -69,106 +73,49 @@ class AmpSocialShare extends AMP.BaseElement {
-      }
-    }
-
-    // Get the anchor or create one.
-    let link = elementByTag(this.element, 'a');
-    if (link == null) {
-      link = this.getWin().document.createElement('a');
-      link.textContent = this.typeConfig_['text'];
-      link.setAttribute('target', '_blank');
-
-      // Get the container or create one.
-      let container = elementByTag(this.element, 'span');
-      if (container == null) {
-        container = this.getWin().document.createElement('span');
-        container.classList.add(CLASSNAME_PREFIX + this.type_);
-
-        // Only add the container to the element if it didn't exist
-        this.element.appendChild(container);
-      }
-      container.appendChild(link);
+    /** @private @const {!Object} */
+    this.params_ = Object.assign({}, this.typeConfig_.defaultParams || {});
+    Object.assign(this.params_, getAttributesAsParams(
+        this.element, attr => {
+          const matches = attr.nodeName.match(/^data-param-(.+)/);
+          return (matches && matches[1]) || null;
+        }));
+
+    const hrefWithVars = addParamsToUrl(this.shareEndpoint_, this.params_);
+    urlReplacementsFor(this.getWin()).expand(hrefWithVars).then(href => {
+      this.element.setAttribute('href', href);
+    });
+    this.element.setAttribute('role', 'link');
+
+    // Install a global click listener only once for all amp-social-share.
+    if (!clickHandlerUnlistenCallback) {
+      this.addGlobalClickListener_();
     }
-
-    // Set share url.
-    link.setAttribute('href',
-      addParamsToUrl(this.typeConfig_['url'], urlParams));
   }
 
   /**
-   * Gets the configuration for the current social element
-   * @param {!element} element
-   * @return {?Object}
+   * Install a global click event listener to handle all amp-social-share targets.
+   * @param win
+   * @private
    */
-  getElementConfig_() {
-    const script = elementByTag(this.element, 'script');
-    let config = {};
-    if (script) {
-      // Get config from script
-      try {
-        config = JSON.parse(script.textContent);
-      } catch (e) {
-        user.error(TAG, 'Malformed JSON configuration. %s', this.element);
-      }
-    } else {
-      // Get config from attributes
-      for (const param in this.typeConfig_['params']) {
-        if (this.element.hasAttribute('data-' + param)) {
-          config[param] = this.element.getAttribute('data-' + param);
-        }
-      }
-    }
-    return config;
+  addGlobalClickListener_() {
+    const win = this.getWin();
+    clickHandlerUnlistenCallback = listenForEventOnTagName(
+        win, 'amp-social-share', 'click', event => {
+          const href = event.target.getAttribute('href');
+          const type = event.target.getAttribute('type');
+          const isDesktop = platformFor(win).isDesktop();
+          if (isDesktop) {
+            const pageViewId = documentInfoFor(win).pageViewId;
+            const popupId = 'share-popup-' + type + pageViewId;
```

Comment:
You mean if we just do `popup`? If that's the case then yeah we can drop the `_blank` branch.